components/match2/wikis/pokemon/match_summary.lua

```diff
@@ -0,0 +1,316 @@
+	      :rightScore(header:createScore(match.opponents[2]))
+	      :rightOpponent(header:createOpponent(match.opponents[2], 'right'))
+
+	return header
+end
+
+function CustomMatchSummary._createBody(match)
+	local body = MatchSummary.Body()
+
+	if match.dateIsExact or (match.date ~= _EPOCH_TIME_EXTENDED and match.date ~= _EPOCH_TIME) then
+		-- dateIsExact means we have both date and time. Show countdown
+		-- if match is not epoch=0, we have a date, so display the date
+		body:addRow(MatchSummary.Row():addElement(
+			DisplayHelper.MatchCountdownBlock(match)
+		))
+	end
+
+	-- Iterate each map
+	for gameIndex, game in ipairs(match.games) do
+		local rowDisplay = CustomMatchSummary._createGame(game, gameIndex, match.date)
+		if rowDisplay then
+			body:addRow(rowDisplay)
+		end
+	end
+
+	-- Add Match MVP(s)
+	local mvpInput = match.extradata.mvp
+	if mvpInput then
+		local mvpData = mw.text.split(mvpInput or '', ',')
+		if String.isNotEmpty(mvpData[1]) then
+			local mvp = MatchSummary.Mvp()
+			for _, player in ipairs(mvpData) do
+				if String.isNotEmpty(player) then
+					mvp:addPlayer(player)
+				end
+			end
+
+			body:addRow(mvp)
+		end
+
+	end
+
+	-- Pre-Process Champion Ban Data
+	local championBanData = {}
+	for gameIndex, game in ipairs(match.games) do
+		local extradata = game.extradata
+		local banData = {{}, {}}
+		local numberOfBans = 0
+		for index = 1, _MAX_NUM_BANS do
+			if String.isNotEmpty(extradata['team1ban' .. index]) then
+				numberOfBans = index
+				banData[1][index] = extradata['team1ban' .. index]
+			end
+			if String.isNotEmpty(extradata['team2ban' .. index]) then
+				numberOfBans = index
+				banData[2][index] = extradata['team2ban' .. index]
+			end
+		end
+
+		if numberOfBans > 0 then
+			banData[1].color = extradata.team1side
+			banData[2].color = extradata.team2side
+			banData.numberOfBans = numberOfBans
+			championBanData[gameIndex] = banData
+		end
+	end
+
+	-- Add the Champion Bans
+	if not Table.isEmpty(championBanData) then
+		local championBan = ChampionBan()
+
+		for gameIndex, banData in ipairs(championBanData) do
+			championBan:banRow(banData, gameIndex, banData.numberOfBans, match.date)
+		end
+
+		body:addRow(championBan)
+	end
+
+	return body
+end
+
+function CustomMatchSummary._createGame(game, gameIndex, date)
+	local row = MatchSummary.Row()
+	local extradata = game.extradata or {}
+
+	local championsData = {{}, {}}
+	local championsDataIsEmpty = true
+	for champIndex = 1, _NUM_CHAMPIONS_PICK do
+		if String.isNotEmpty(extradata['team1champion' .. champIndex]) then
+			championsData[1][champIndex] = extradata['team1champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		if String.isNotEmpty(extradata['team2champion' .. champIndex]) then
+			championsData[2][champIndex] = extradata['team2champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		championsData[1].color = extradata.team1side
+		championsData[2].color = extradata.team2side
+	end
```

Comment:
```suggestion
	local getChampsForTeam = function(team)
		local getChampFromIndex =  function(champIndex)
			return champIndex, String.nilIfEmpty(extradata['team' .. team .. 'champion' .. champIndex])
		end
		return Table.map(Array.range(1, _NUM_CHAMPIONS_PICK, getChampFromIndex))
	end

	local championsData = {getChampsForTeam(1), getChampsForTeam(2)}
	local championsDataIsEmpty = Table.isNotEmpty(championsData[1]) or Table.isNotEmpty(championsData[2])
	championsData[1].color = extradata.team1side
	championsData[2].color = extradata.team2side
```
Curious if this would work, as I've never been a fan of that loop

Reply:
@hjpalpha Can you double check the logic as well?

Reply:
think it should work

Reply:
have tested, by adding any that is already marked resolved here a popup errors out, with these in the log 
![image](https://user-images.githubusercontent.com/88981446/182397116-738645db-3a64-47e0-b2b3-6444c7e893d4.png)

Reply:
Change `	local championsDataIsEmpty = Array.all(Table.isEmpty)` to `	local championsDataIsEmpty = Array.all(championsData, Table.isEmpty)`

Reply:
> Change ` local championsDataIsEmpty = Array.all(Table.isEmpty)` to ` local championsDataIsEmpty = Array.all(championsData, Table.isEmpty)`

noted, after changing this line. Another set of log errors pop 
![image](https://user-images.githubusercontent.com/88981446/182406141-22e5ba0f-3bc2-43b6-bb8e-be4c74986f0d.png)

Reply:
It's unrelated to this change (most likely)